docs/source/rfc/0001-data-authorization-middleware.md

```diff
@@ -0,0 +1,628 @@
+- Feature flag controlled rollouts
+- Comprehensive monitoring and alerting
+
+### Compatibility
+
+**Backward Compatibility:**
+
+- No changes to existing API contracts
+- All current authentication methods preserved
+- VPD remains active during migration
+- Gradual feature rollout with rollback capability
+
+**Forward Compatibility:**
+
+- PostgreSQL migration ready
+- Cloud-native architecture
+- Standard protocols (OAuth2, OIDC, REST)
+- Extensible policy framework
+
+**Migration Compatibility:**
+
+- Parallel operation with existing VPD
+- Office-by-office migration capability
+- Emergency fallback to VPD-only mode
+- Data consistency validation tools
+
+### User Impact
+
+**End Users (Zero Impact):**
+
+- Same API endpoints and authentication
+- No client application changes required
+- Transparent authorization enhancement
+
+**Data Administrators:**
+
+- New web-based access management UI
+- Enhanced granular access controls
+- Real-time policy updates
+- Improved audit and reporting capabilities
+
+**Operations Team:**
+
+- New service to monitor and maintain
+- Enhanced security posture
+- Better troubleshooting capabilities
+- Standardized policy management
+
+### Local Development & Testing Plan
+
+1. **Local Environment Setup**
+
+   - Deploy using Docker/Podman Compose
+   - Configure all services (Database, API, Auth Service, OPA)
+   - Load sample timeseries test data
+   - Verify end-to-end connectivity
+
+2. **Timeseries Authorization Testing**
+
+   - Implement authorization for timeseries endpoints
+   - Test all 7 user personas with timeseries data
+   - Validate embargo rules and time-based constraints
+   - Performance benchmarking in local environment
+
+3. **Integration Validation**
+
+   - Test Authorization Helper Library with timeseries controllers
+   - Validate JWT pass-through and context headers
+   - Ensure VPD compatibility during transition
+   - Document integration patterns for future resources
+
+## Detailed Design
+
+### Implementation Strategy
+
+**Transparent Proxy Pattern:**
+
+- Authorization layer for `/cwms-data/timeseries/*` endpoints
+- Policy-based access control with intelligent caching
+- Single structured header for authorization context
+- Selective response filtering for embargo and persona rules
+
+**Java API Integration:**
+
+- Authorization Helper Library for timeseries controllers
+- Minimal changes to TimeSeriesController (2-3 lines)
+- Structured authorization context via single HTTP header
+- Focus on demonstrating pattern for future resources
+
+### Java API Integration Approach
+
+**Authorization Helper Library Pattern:**
+
+```java
+@RestController
+public class TimeSeriesController {
+    @Autowired
+    private CwmsAuthorizationHelper authHelper;
+
+    @GetMapping("/timeseries")
+    public ResponseEntity<List<TimeSeries>> getTimeSeries(
+            String office, Context ctx) {
+
+        QueryBuilder query = QueryBuilder.create().table("cwms_ts_data");
```

Comment:
The table information would either not be known at this point or possibly meaningless.

What CDA does know is the endpoint URL and the DTOs involved. I would suspect that would be able to provide the required information checks.

Doesn't negate the point the sample is making though, same concept different source.